### ISSUE TITLE
Build failed

### ISSUE THREAD

**Issue by jcoonrod** [dahliaOS/buildroot#15]
**Describe the bug**
I followed the instructions to try a build today with the current master

**What kind of a bug is it?**
Make halted
**To Reproduce**
1. Followed instructions through the make command.

**Expected behavior**
Expected a build
**Screenshots**
Halted with these error..
 CC       libgdk_3_la-gdkmarshalers.lo
make[6]: *** No rule to make target 'Gdk-3.0.typelib', needed by 'all-am'.  Stop.
make[6]: *** Waiting for unfinished jobs....
  CC       libgdk_3_la-gdkresources.lo
make[5]: *** [Makefile:1663: all-recursive] Error 1
make[4]: *** [Makefile:1092: all] Error 2
make[3]: *** [Makefile:746: all-recursive] Error 1
make[2]: *** [Makefile:619: all] Error 2
make[1]: *** [package/pkg-generic.mk:270: /home/john/buildroot/output/build/libgtk3-3.24.30/.stamp_built] Error 2
make: *** [Makefile:84: _all] Error 2

**Desktop (please complete the following information):**
 - OS: Ubuntu 20
 - Device: [Asus PN41 mini pc

**Additional context**

**Comment by nmcain** [comment]
Are the GTK development libraries installed?

**Comment by nmcain** [comment]
For clarification, this is an issue with upstream GTK, it appears gobject-introspection is missing on the host device, as far as I can tell.

**Comment by quintenvandamme** [comment]
If you use `dart compile exe bin/pkg.dart` it generates a file called pkg.exe. In your case you don't have dart installed and thats why it can't build and find pkg.exe. For more info see [https://dart.dev/](https://dart.dev/tools/dart-compile).

**Comment by jcoonrod** [comment]
but is that something I can control or do I need to change your build files?

**Comment by nmcain** [comment]
The dap package can be disabled in menu config, `make menuconfig`, go to target packages - dahliaOS applications - and unselect dap

**Comment by nmcain** [comment]
I'd certainly like to make the switch at some point, not sure how complete the rust coreutils are at this point, though.

**Comment by jcoonrod** [comment]
Oops - one more pre-requesite needed. /bin/bash: flutter: command not found
make[1]: *** [package/pkg-generic.mk:270: /home/john/buildroot/output/build/pangolin_desktop/.stamp_built] Error 127
make: *** [Makefile:84: _all] Error 2

Try try again!
Oops - this one has me baffled - do I need to delete stuff before restarting make?
fatal: destination path 'pangolin_desktop' already exists and is not an empty directory.
make[1]: *** [package/pkg-generic.mk:270: /home/john/buildroot/output/build/pangolin_desktop/.stamp_built] Error 128
make: *** [Makefile:84: _all] Error 2

**Comment by jcoonrod** [comment]
Ok - that sort of worked, then this happened: [   +4 ms] "flutter linux" took 2,668ms.
[   +4 ms] Error: No pubspec.yaml file found.
           This command should be run from the root of your Flutter project.
[        ]

**Comment by nmcain** [comment]
From here, when the pangolin step errors out, you'll need to `rm -rf output/build/pangolin*`, the script is not written to clean itself up.

**Comment by jcoonrod** [comment]
Yes, I did that - it led to some progress, but what do I do about the missing yaml file?

**Comment by nmcain** [comment]
The missing yaml file should not be causing a problem, you can try `rm -rf dl/pangolin*&&rm -rf output/build/pangolin*`

**Comment by jcoonrod** [comment]
Ok, had to issue another pre-requisite command:
 flutter config --enable-linux-desktop 
That let make get it to go pretty close to the end, and then it failed with 
make[1]: *** [fs/cpio/cpio.mk:75: /home/john/buildroot/output/images/rootfs.cpio] Error 137
make: *** [Makefile:84: _all] Error 2
Looked this up - an "Out of memory" error (I just have 4gig - will buy more tomorrow!)

**Comment by nmcain** [comment]
Yeah, this process is significantly resource intensive, but it looks like your build completed successfully and has moved on to image packaging. The CPIO image is only required for the legacy image - if you just want an EFI image, you can turn off CPIO and ISO9660 in filesystem options, I'm not sure how the BTRFS image tool works in low-memory environments. Thanks for making this issue, it will be a very helpful reference in making our builds completely reproducible :)

**Comment by jcoonrod** [comment]
Hello! OK! Installed more memory and the make completed with an iso and an img file in buildroot/output/images/dahliaOS - I used dd to create a bootable USB stick and it did a panic after finding no init. What should I try next
![PXL_20220213_205104607](https://user-images.githubusercontent.com/54367582/153774769-6d47be3e-5ba1-451b-9019-587a0de4ffeb.jpg)
?

**Comment by larsb24** [comment]
Try source ~/.profile

**Comment by nmcain** [comment]
On the build machine, check that an init system is selected under target options

**Comment by jcoonrod** [comment]
I could not find that option in the config manager - Target options were just machine types like x64...
